data-mongodb/src/main/java/io/micronaut/data/mongodb/operations/AbstractMongoRepositoryOperations.java

```diff
@@ -168,8 +168,10 @@ protected <R> R convertResult(CodecRegistry codecRegistry,
             try {
                 return mapIntrospectedObject(result, resultType);
```

Comment:
What happens here is we get this map
![Screenshot 2024-12-09 at 1 19 57 PM](https://github.com/user-attachments/assets/65a59459-ca1f-4e8b-9ac0-b2fd17b3f788)
which cannot be deserialized to the record 
```
record CustomerView(@BsonId @BsonRepresentation(OBJECT_ID) String id, String firstName, String lastName) {
}
```
because `_id` and `id` are mismatched in introspected mapper. Therefore we catch exception and fallback to this code 
```
else if (isDtoProjection) {
            R dtoResult = MongoUtils.toValue(result.asDocument(), resultType, codecRegistry);
            if (resultType.isInstance(dtoResult)) {
                return dtoResult;
            }
            return conversionService.convertRequired(dtoResult, resultType);
        }
```